.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 The StableHLO Authors.
+
+# The setup-build action gets everything needed by buildAndTest into the workspace.
+# This checks out the proper revision of the LLVM repository, installs ninja, and sets
+# up the ccache for faster re-compilation.
+name: "Setup build environment (ninja, ccache, llvm, lld)"
+
+inputs:
+  llvm-ref:
+    description: |
+      LLVM Commit Ref for checkout and build. Used for ccache value and checkout.
+    required: true
+
+
+runs:
+  # Based on https://github.com/llvm/torch-mlir/blob/main/.github/actions/setup-build/action.yml
+  using: "composite"
+
+  steps:
+  # Checkout llvm at revision specified in input argument.
+  - uses: actions/checkout@v2
+    with:
+      repository: llvm/llvm-project
+      ref: ${{ inputs.llvm-ref }}
+      path: llvm-project
+      fetch-depth: 1
+
+  # Get ninja for cmake build.
+  - name: Install Ninja
+    uses: llvm/actions/install-ninja@55d844821959226fab4911f96f37071c1d4c3268
+
+  # Get LLD
```

Comment:
Perhaps expand this comment to explain why this is needed?

Reply:
Added comment "Improves build speed on Linux" - that's the main reason to use LLD ([ref]( https://mlir.llvm.org/getting_started/)). The MHLO build script requires LLD in it's current state so this helps keep those in line, but that's a minor reason since this uses the `ci_build_*` scripts.